### DISCUSSION TITLE
JsonConvert<PluginExecutionContext> gives error

### DISCUSSION THREAD

**Discussion by rasotoipc** [microsoft/PowerPlatform-DataverseServiceClient#77] [Q&A]
I have created an Azure Aware plugin and have passed the PluginExecutionContext to the the service end point (webhook). The function recives the plugin execution context where I am desearlizing with the PluginExuctionContext in this library.

Get error:

Cannot create and populate list type Microsoft.Xrm.Sdk.ParameterCollection. Path 'InputParameters', line 1, position 283.

![2020-12-07_12-51-06](https://user-images.githubusercontent.com/36775149/101402662-0aae8900-38a2-11eb-8cf4-f2932040cceb.jpg)
![2020-12-07_12-51-37](https://user-images.githubusercontent.com/36775149/101402665-0b471f80-38a2-11eb-964d-3dcbc8c119eb.jpg)

**Comment by BetimBeja** [comment]
I would recommend the following repository [XrmEntitySerializer](https://github.com/albanian-xrm/Xrm-Entity-Serializer) to solve your problem.

**Comment by JimDaly** [comment]
Can you do this with the current SDK?
The value sent is a RemoteExecutionContext: https://docs.microsoft.com/en-us/dotnet/api/microsoft.xrm.sdk.remoteexecutioncontext?view=dynamics-general-ce-9

**Comment by BetimBeja** [comment]
There are some tricks being used to make sure the serialization/deserialization is possible. So it depends...

**Comment by BetimBeja** [comment]
@JimDaly 
I just pushed some changes so that it is possible to [deserialize](https://github.com/albanian-xrm/Xrm-Entity-Serializer/blob/6990ac6921d12a0416ab42d27d6b5585172c255e/XrmEntitySerializer.Tests/RemoteExecutionContextTests.cs#L39-L58) the RemoteExecutionContext. 
I used some example string from [this blog](https://www.inogic.com/blog/2018/06/parse-json-string-that-represents-the-dynamics-365-plugin-execution-context-received-in-azure-function/)

**Comment by JimDaly** [comment]
So, can I close this?

**Comment by BetimBeja** [comment]
@rasotoipc does the latest release of [XrmEntitySerializer.Core](https://www.nuget.org/packages/XrmEntitySerializer.Core) solve your problem?

**Comment by BetimBeja** [comment]
@rasotoipc I developed another package [AlbanianXrm.PowerPlatform.EntitySerializer](https://github.com/albanian-xrm/PowerPlatform-Entity-Serializer) without the Newtonsoft.Json dependency (it is based on System.Text.Json), it is not very stable yet but might be helpful.

**Comment by MattB-msft** [comment]
Folks... Moved this to the discussion forum as it relates to the plugin infrastructure more then the core SDK. 
We are still following it however :)
